### PR TITLE
dev: Fix snap permissions errors for ttn-lw-cli

### LINKED ISSUE #1670: ttn-lw-cli snap can't read config file
#### Summary
ttn-lw-cli installed on via snap can't open the config file in ~/.ttn-lw-cli.yml, it errors with
```
$ ttn-lw-cli config         
open /home/al/.ttn-lw-cli.yml: permission denied
```
I've followed the installation guide here:
https://enterprise.thethingsstack.io/v3.3.1/guides/getting-started/installation/#command-line-interface-optional
and configuration here (modified for my instance)
https://enterprise.thethingsstack.io/v3.3.1/guides/getting-started/configuration/#command-line-interface

I can make ttn-lw-cli work passing the config file with -c, but only if the config file is named something else:
```
$ ttn-lw-cli config           
open /home/al/.ttn-lw-cli.yml: permission denied

$ ttn-lw-cli -c .ttn-lw-cli.yml config 
open .ttn-lw-cli.yml: permission denied

$ ttn-lw-cli -c ttn.yml config        
                         --allow-unknown-hosts="false"
                  --application-server-enabled="true"
 ...
```
My config file is good, and works when renamed and passed with -c. snap list shows:
```
ttn-lw-stack        3.3.1                       94    stable    rvolosatovs-ttn  -
```

#### Steps to Reproduce

1. Install the cli per the docs above
2. Add a .ttn-lw-cli.yml and see the permission denied error when running the cli

#### What do you see now?

...

#### What do you want to see instead?

...

#### Environment

Ubuntu 18.04

```
The Things Network Command-line Interface: ttn-lw-cli
Version:             3.3.1
Build date:          2019-11-26T12:36:02Z
Git commit:          8bedf3f4e
Go version:          go1.13.4
OS/Arch:             linux/amd64

```

#### How do you propose to implement this?

...

#### Can you do this yourself and submit a Pull Request?

...

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #1670 

#### Changes

- add `personal-files` plugin, with read permissions for `.ttn-lw-cli.yml` and `.ttn-lw-stack.yml` files

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [X] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [X] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -115,11 +115,16 @@ snapcrafts:
     grade: stable
     confinement: strict
     publish: true
+    plugs:
+      personal-files:
+        read:
+        - $HOME/.ttn-lw-stack.yml
```
Comment (rvolosatovs) [inline .goreleaser.yml]: Is this a thing? I don't think I ever saw this configuration file mentioned anywhere and I'm unable to find it in the codebase either. @htdvisser ?
I think we always suggest configuring the stack via environment variables.

File: .goreleaser.yml [hunk 2]
```diff
@@ -115,11 +115,16 @@ snapcrafts:
...
+        - $HOME/.ttn-lw-cli.yml
```
Comment (rvolosatovs) [inline .goreleaser.yml]: Currently we provide 3 different ways to configure the configuration file: https://github.com/TheThingsNetwork/lorawan-stack/blob/19b035b63739d3c0f562927b07a1475bf36d13c8/doc/content/guides/getting-started/cli/_index.md#L37-L39
This only supports one of them in a very specific location.

1. Is it possible to support all 3 configuration options (i.e. env variable pointing to arbitrary location, cli flag pointing to arbitrary location, config in `XDG_CONFIG_HOME` or working directory). Maybe there's a way the user can configure the allowed paths locally?
2. If (1.) is not possible, then we should at least support the other 2 possible locations as well - in `XDG_CONFIG_HOME` and in working directory.

File: .goreleaser.yml (hunk 1 above)
Comment (htdvisser) [reply .goreleaser.yml]: This is just how our config package works. It reads `.<binary>.yml` from the workdir, homedir and (xdg) config dir.

File: .goreleaser.yml (hunk 2 above)
Comment (neoaggelos) [reply .goreleaser.yml]: The issue only happens for `$HOME/.ttn-lw-cli.yml`, permissions for the working directory and XDG_CONFIG_HOME are OK as-is.

File: .goreleaser.yml (hunk 1 above)
Comment (neoaggelos) [reply .goreleaser.yml]: Though the documentation suggestions environment variables, the stack does search for config in `.ttn-lw-stack.yml`

**Comment by GillesC** [comment]
I still get the permission denied on the yml file even though I have the latest stable version. Same as issue #1670.

Output of `snap info ttn-lw-stack`:
```
name:      ttn-lw-stack
summary:   The Things Stack for LoRaWAN
publisher: The Things Network (rvolosatovs-ttn)
store-url: https://snapcraft.io/ttn-lw-stack
contact:   stack@thethingsnetwork.org
license:   unset
description: |
  The Things Stack for LoRaWAN
commands:
  - ttn-lw-stack.ttn-lw-cli
  - ttn-lw-stack
snap-id:      Qt4j7Cz4cqDwDZ8OHEvh1N3gEN9VzfSf
tracking:     latest/stable
refresh-date: today at 09:13 UTC
channels:
  latest/stable:    3.17.2 2022-01-30 (476) 25MB -
  latest/candidate: 3.17.2 2022-01-30 (476) 25MB -
  latest/beta:      3.17.2 2022-01-30 (476) 25MB -
  latest/edge:      3.17.2 2022-01-30 (476) 25MB -
installed:          3.17.2            (476) 25MB -
```

Error:
```bash
ttn-lw-cli use eu1.cloud.thethings.network
INFO	Configuring for The Things Stack Community Edition	{"cluster_id": "eu1"}
INFO	Set OAuth Server address	{"address": "https://eu1.cloud.thethings.network/oauth"}
INFO	Set Identity Server gRPC address	{"address": "eu1.cloud.thethings.network:8884"}
error:cmd/ttn-lw-cli/commands:fail_write (failed to write `.ttn-lw-cli.yml`)
    file=.ttn-lw-cli.yml
    correlation_id=597968d32fd6425bbcb90cb8859bf2a3
--- open .ttn-lw-cli.yml: permission denied

```

**Pull request merged**